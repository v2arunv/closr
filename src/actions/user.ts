import {IUser} from "../models/users";

export interface IUserAction {
    type: string,
    user: IUser,
}

export const setUser = (user: IUser): IUserAction => {
    return <IUserAction> {
        type: 'SET_USER',
        user: user,
    }
};