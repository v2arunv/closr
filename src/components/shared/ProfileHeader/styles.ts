import {StyleSheet} from "react-native";
import common from '@common/styles';

const styles = StyleSheet.create({

    header: {
        flexDirection: 'column',
        justifyContent: 'flex-start',
        alignItems: 'flex-start',
        backgroundColor: '#ffffff',
        padding: 10,
    },
    profilePictureContainer: {
        marginBottom: 10
    },
    picture: {
        width: 100,
        height: 100,
        borderRadius: 50,
    },
    nameContainer: {
        flex: 1,
    },
    name: {
        fontWeight: 'bold',
        fontSize: 32,
    },
    username: {
        fontWeight: '300',
        color: common.subtextColor,
    }
});

export default styles