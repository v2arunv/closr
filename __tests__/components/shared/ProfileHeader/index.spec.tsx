import React from 'react';
import ProfileHeader from '@sharedComponents/ProfileHeader';
import renderer from 'react-test-renderer';
import user from '@mocks/users';

test('renders correctly', () => {
    const tree = renderer.create(<ProfileHeader user={user}/>).toJSON();
    expect(tree).toMatchSnapshot();
});