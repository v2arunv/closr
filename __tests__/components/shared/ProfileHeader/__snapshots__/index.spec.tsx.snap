// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#ffffff",
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "padding": 10,
      },
      Object {
        "borderBottomColor": "#ddd",
        "borderBottomWidth": 0,
        "elevation": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
    ]
  }
>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <Image
      defaultSource={
        Object {
          "testUri": "../../../src/assets/images/dp_placeholder.jpg",
        }
      }
      source={
        Object {
          "uri": "http://www.sclance.com/pngs/image-placeholder-png/image_placeholder_png_698152.png",
        }
      }
      style={
        Object {
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 32,
          "fontWeight": "bold",
        }
      }
    >
      Leanne Graham
    </Text>
    <Text
      style={
        Object {
          "color": "#505050",
          "fontWeight": "300",
        }
      }
    >
      @
      Bret
    </Text>
  </View>
</View>
`;
